api/src/main/resources/workbench.yaml

```diff
@@ -1164,6 +1164,25 @@ paths:
           schema:
             $ref: "#/definitions/ConceptSet"
 
+# Concepts ###############################################################################
+
+  /v1/workspaces/{workspaceNamespace}/{workspaceId}/domainInfo:
```

Comment:
Hmm... should this be under workspaces? Couldn't you theoretically need to use this outside of a workspace context?

Reply:
1) One reason I put this (and workbench concept search) under workspace is to allow for in future changes that would be workspace-specific (but are currently out of scope)... particularly, returning counts for concepts matching participants in a cohort, rather than the entire CDR.

2) The UX for data browser and concept search is similar, but not the same (at least right now). For example, the workbench concept search is going to let you choose whether or not to include non-standard concepts in the results... that's why we need both standard and non-standard counts in the results. We also explicitly aren't handling PPI questions specially right now. The API responses are going to be different as a result. I'm planning on sharing backend code and API code where it makes sense, and writing new code when necessary. We shouldn't try to make the code identical when the UX doesn't match... that will just make life more confusing.

Reply:
I agree we don't need to make code identical. I think first though, apis should return the same types when they can.  IE concept, domain_info, analysis ... The backend apis can do whatever -- be different, apply to cohort or not but just return common types where it makes sense. And we may need to design some common types.   

Re: 1) I think I still would want that you return the same type of object to the ui weather the domain or concept was relative to a cohort or not.  I think of all our counts now as applying to a cohort. Cohort 0 -- everyone.  You can save the results for that , then run on another cohort and display the comparison -- like 55 people in your cohort out of the 255 people in the whole cohort have this disease. 

Re: 2 -- non standard counts is something that the databrowser wouldn't mind having. PPI questions is something the concept set builder wouldn't mind having. Bad example maybe? :) At this stage of development I'd say a common api for these things  and let the ui's just filter out what they don't want. I'm partial to building out apis and let the ui decide what it wants to use.  Especially when it saves work or does not cost anything or simplifies a code base or you are kind of thinking you are going to use it in the future and are already in that code and can just do it now real easy. An overbuilt api in early development for count data is not a bad thing IMO as security is not a concern.  

This is why you see some api call that are not used like domainFilters and getSurveys ...

Reply:
Re: #1, that's fine... I will attempt to have the same data types used by data browser and this search. This may involve some re-working of things.

Re: #2 -- I think we have a philosophical disagreement here about dead code. :) From my vantage point, it's much better to delete  code that isn't being used in general at all points in the development process. Dead code confuses developers and requires maintenance (even though it isn't actually doing anything.) Maintaining extra REST API fields / objects / methods is extra problematic, because it becomes hard to tell at some point whether something is in use somewhere in a UI or client library, and hence whether removing it would break things. WRT security: we are fast approaching having a prod CDR, at which point we'll need to take security almost as seriously as we will at external launch... extra code hanging around makes that harder.

I can see the point about *sharing* API code even when only a subset of the data is currently being used by a given client, particularly when we expect the data to be used in future... but we need to be careful not to do this if things diverge much, particularly when there's a significant performance hit. I can imagine all kinds of ways of surfacing data in the data browser and workbench, and until we actually pick one, I don't know what data would be useful where. (For example: it may be that we're going to have a different section of the Concepts tab for PPI and physical measurements in future; and we may have a different section of the data browser for non-standard concepts.)